### PR TITLE
Bug/#167 multiple discards of the same record

### LINKED ISSUE #167: Cannot read property '_$onMessage' of undefined
```javascript
/deepstream.io-client-js/src/record/record-handler.js:276
	var onMessage = this._records[ recordName ]._$onMessage.bind( this._records[ recordName ] );
	                                           ^

TypeError: Cannot read property '_$onMessage' of undefined
    at RecordHandler._onDestroyPending (foo/node_modules/deepstream.io-client-js/src/record/record-handler.js:276:45)
    at Emitter.emit (foo/node_modules/component-emitter/index.js:134:20)
    at .<anonymous> (foo/node_modules/deepstream.io-client-js/src/record/record.js:236:9)
    at on (foo/node_modules/component-emitter/index.js:66:8)
    at Emitter.emit (foo/node_modules/component-emitter/index.js:134:20)
    at Record._setReady (foo/node_modules/deepstream.io-client-js/src/record/record.js:455:7)
    at Record._onRead (foo/node_modules/deepstream.io-client-js/src/record/record.js:439:7)
    at Record._$onMessage (foo/node_modules/deepstream.io-client-js/src/record/record.js:289:9)
    at RecordHandler._$handle (foo/node_modules/deepstream.io-client-js/src/record/record-handler.js:228:25)
    at Client._$onMessage (foo/node_modules/deepstream.io-client-js/src/client.js:130:42)
```
```

### PR DESCRIPTION
**Pull request by timaschew**
fix #167

### CODE REVIEW

File: test-e2e/specs/recordSpec.js [hunk 1]
```diff
@@ -241,6 +241,24 @@ describe( 'record', function() {
 		}, 20 );
 	});
 
+	it( 'create and discards two times in a row', function( done ) {
```
Comment (yasserf) [inline test-e2e/specs/recordSpec.js]: did this fail before the fix?

**Comment by yasserf** [comment]
ci is still broken =(

File: test-e2e/specs/recordSpec.js (hunk 1 above)
Comment (timaschew) [inline test-e2e/specs/recordSpec.js]: it failed before

**Comment by timaschew** [comment]
Oh, yeah that's because coverage is doing it within one command.
We can't split it into two commands for the coverage 😢

**Pull request merged**